### PR TITLE
Allow conversion to DynapcnnNetwork for modules with Identity

### PR DESCRIPTION
**Pull request by bauerfe**
Addresses #201 . 
When converting a Sequential module to a list, it now automatically skips layers such as Identity or Dropout.

Ignoring these layers in the model building routine, as has been done previously falls short in some cases, when the layer to be ignored sits between two pooling layers or a conv layer and a pooling layer.

### CODE REVIEW

File: sinabs/backend/dynapcnn/utils.py [hunk 1]
```diff
@@ -409,8 +410,8 @@ def build_from_list(
         in_shape = dvs_layer.get_output_shape()
     # Find and populate dynapcnn layers
     while lyr_indx_next < len(layers):
-        if isinstance(layers[lyr_indx_next], (nn.Dropout, nn.Dropout2d, nn.Flatten)):
-            # - Ignore dropout and flatten layers
+        if isinstance(layers[lyr_indx_next], (nn.Identity, nn.Dropout, nn.Dropout2d, nn.Flatten)):
```
Comment (bauerfe) [inline sinabs/backend/dynapcnn/utils.py:413]: We might consider dropping this if-statement completely, because now these types should already have been filtered out by the `convert_model_to_layer_list` function. However, we can also leave it in here as a safety net.

File: sinabs/backend/dynapcnn/utils.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 if TYPE_CHECKING:
     from sinabs.backend.dynapcnn.dynapcnn_network import DynapcnnNetwork
 
+DEFAULT_IGNORED_LAYER_TYPES = (nn.Identity, nn.Dropout, nn.Dropout2d, nn.Flatten)
```
Comment (sheiksadique) [inline sinabs/backend/dynapcnn/utils.py:18]: For Dropout layers, don't we have to account for parameter rescaling? Is this handled somewhere?

File: sinabs/backend/dynapcnn/utils.py (hunk 1 above)
Comment (sheiksadique) [reply sinabs/backend/dynapcnn/utils.py:413]: I agree, but perhaps to ensure that this method is usable independently, it is useful to retain it. 

We could also use the `DEFAULT_IGNORED_LAYER_TYPES` here I assume? Do you foresee any potential issues in doing that?

File: sinabs/backend/dynapcnn/utils.py (hunk 2 above)
Comment (bauerfe) [reply sinabs/backend/dynapcnn/utils.py:18]: Until now dropout layers have been simply ignored at conversion. This PR does not change that. I would argue that this is the correct behavior. Because of the way dropout is implemented in PyTorch, there should be no scaling issue: During training, those outputs that are not dropped are scaled up accordingly. During inference nothing is dropped and nothing is scaled.

File: sinabs/backend/dynapcnn/utils.py (hunk 1 above)
Comment (bauerfe) [reply sinabs/backend/dynapcnn/utils.py:413]: That's a good suggestion, thank you!

Comment (bauerfe) [reply sinabs/backend/dynapcnn/utils.py]: Updated in the latest commit.
